src/aui-form-validator/js/aui-form-validator.js

```diff
@@ -1324,7 +1324,9 @@ var FormValidator = A.Component.create({
             var skipValidationTargetSelector = instance.get('skipValidationTargetSelector');
 
             if (!event.relatedTarget || !event.relatedTarget.getDOMNode().matches(skipValidationTargetSelector)) {
-                instance.validateField(event.target);
+                skipValidationTargetSelector.forEach(function(item) {
+                    instance.validateField(event.target);
+                  });
```

Comment:
Something does not seem right about this code.

So we have an array of elements, then shouldn't we be looping over the items in that array when we are checking if the `getDOMNode().matches()` instead of just looping over the number of items in the array when we run `validateField`?  

Additional, we aren't even passing the array elements to the `validateField()` method, but instead we are just looping over the number of elements in the array, so I don't really see how this would solve the issue or provide any benefit to the code.

Can you explain?

Reply:
Hi @jonmak08,
Thanks for your review again!
I have checked and you are right, I don't need to iterate inside the if statement and I didn't use `item` at all.
I initially thought `matches` would only accept a string, and not an array, so that's why I introduced the `forEach`, but it's unnecessary, if we just check the values of the Array through `getDOMNode().matches(skipValidationTargetSelector)` we get the right boolean value:

```
event.relatedTarget.getDOMNode().matches(skipValidationTargetSelector)
true
event.relatedTarget
Y_Node {_yuid: "yui_patched_v3_18_1_1_1567160663044_28822", _node: input#_com_liferay_journal_web_portlet_JournalPortlet_useSmallImage.field, _stateProxy: input#_com_liferay_journal_web_portlet_JournalPortlet_useSmallImage.field, _plugins: {…}}

```

I checked the behaviour with this code and it works as expected: the values of the skippable target nodes don't fire validation, however, other events like clcicking on 'Publish' would result in validateField.

What are your thoughts on the above? I'll send you a new PR soon.

Kind regards,
Orsolya

Reply:
Hi @orsolyalukacs,

I am trying to understand what you are saying.  The passed argument to `validateField` is `event.target`, but the element that is being checked to see if it `matches` is `event.relatedTarget`.  Should we be passing `event.relatedTarget` to the `validateField` method instead?

I think that is what you are saying, but let me know if I am missing something.

Thanks!

Reply:
Hi @jonmak08,
We are checking the right values.  We need to use `matches` for `event.relatedTarget ` to see if we have the node in the event (the checkbox or the cancel-button), so we can skip the validation, and then we need to validate `event.target`, because that's the title-field in the Web Content.

Reply:
Hi @orsolyalukacs,

In looking at what you sent and posted, is the below behavior what is expected and intended?

![bug](https://user-images.githubusercontent.com/860951/64292516-695ab980-cf1f-11e9-82a8-fae3adaf1ecd.gif)

I am slightly hesitant to push this through because this change will default all checkboxes to skip validation.  Is there no way to pass this `attribute` through in the portal's edit journal page instead of through AUI directly?

Aren't we calling aui-form-validator inside portal's journal page somewhere and can't we just pass this attribute through to AUI?

Please let me know if you have any questions. Thanks!

Reply:
Hi @jonmak08,

Thanks for looking into this. 

Exactly, the second video shows the desired behaviour: when we click on the checkbox at 'Use Small Image', the checkbox gets checked and the whole Small Image div expands immediately, so the customer doesn't have to click multiple times as you captured in the first recording.

I think you're right, it wouldn't be necessary to skip validation for all the ckeckboxes by all means, so I'll try changing the journal-web inide the portal modules as you suggested.

Kind regards,

Reply:
Hey @orsolyalukacs,

I would first check to see if the `Use Small Image` check box has any validation associated to it.  If it does, we could probably just remove that if the validation is not required from within journal-web.  If the checkbox does not currently have any validation, I think the discussion would have to be with the journal web component team to determine if autofocusing on the title field to begin is the desired behavior since validation occurs on blur of a required text field which is why the title field error message is in focus when you click elsewhere on the page.

I hope this helps.  Please let me know if you have any questions. Thanks!

Reply:
Hi @jonmak08,

Thanks for your help.
I've been investigating what you suggested above and found that there is no validation on the checkbox at the moment. The title field is in autofocus when we load the page, and if left empty, we get an error message inserted in the html: "This field is required."
After that, the click event doesn't fire any click event, and so leaves our checkbox unchecked and the div unexpanded. 

I checked what happens if we do not set autofocus on the title field, and it resolves the first event after pageload: clicking inside the checkbox works fine, and the div gets expanded. 

![Screen Recording 2019-09-10 at 15 21 36](https://user-images.githubusercontent.com/25482492/64617779-53e20580-d3df-11e9-804c-a01b748f04d5.gif)

However, when we have the small-image container expanded meanwhile we click inside the title field, then go back and try to close the small-image container by clicking the checkbox, it doesn't close, since the validation fires this time, instead of on pageload.

We've also observed that the same happens if we try vanilla javascript in the same scenario: whenever there is a content change in the dom, such as innerHTML, the validation for another checkbox won't bubble up. It's because events `focus` and `blur` do not bubble.

My question is: if not setting autofocus on the title field, can we change the validation flow of events somehow or is it a limitation?

Reply:
Hi @orsolyalukacs,

You are right about the `blur` event and the `focus` event not bubbling.  There is a `focusin` and `focusout` event that bubbles.  We might be able to use that event in case you want to capture the bubbling.

See https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event and https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event.